contracts/core/OperationExecutor.sol

```diff
@@ -152,15 +153,15 @@ contract OperationExecutor is IERC3156FlashBorrower, IFlashLoanRecipient {
   ) external override {
     address asset = address(tokens[0]);
     address lender = registry.getRegisteredService(BALANCER_VAULT);
-    FlashloanData memory flData = abi.decode(data, (FlashloanData));
+    FlashloanWithInitiatorData memory flData = abi.decode(data, (FlashloanWithInitiatorData));
 
     checkIfLenderIsTrusted(lender);
     checkIfFlashloanedAssetIsTheRequiredOne(asset, flData.asset);
     checkIfFlashloanedAmountIsTheRequiredOne(asset, flData.amount);
 
     // Can we just assume that the initiator is the OperationExecutor?
     // Are there any security consequences because of that assumption?
-    processFlashloan(flData, registry.getRegisteredService(OPERATION_EXECUTOR));
```

Comment:
That was completely wrong indeed. It worked on my case because I was executing the flow directly and not through a proxy.  Indeed the proxy is the initiator. Can you check the test for L2 though ? They might be broken and if some, let me know and I can work on them